monitor/services/src/main/java/org/datacleaner/monitor/server/components/InputColumnAndMappedPropertyRewriter.java

```diff
@@ -45,58 +47,138 @@ public boolean rewriteInput(TransformerDescriptor<?> transformer, ProcessStatele
-                        // second mapped property for the same inputColumn property.
-                        // (Is this an error?)
-                        return false;
+        if(inputColumnProp != null) {
+            // Check if there is a mapped property for the input column.
+            for (ConfiguredPropertyDescriptor prop : props) {
+                MappedProperty mappedPropAnnot = prop.getAnnotation(MappedProperty.class);
+                if(mappedPropAnnot != null) {
+                    if(inputColumnProp.getName().equals(mappedPropAnnot.value())) {
+                        if(mappedProp != null) {
+                            // second mapped property for the same inputColumn property.
+                            // (Is this an error?)
+                            return false;
+                        }
+                        mappedProp = prop;
                     }
-                    mappedProp = prop;
                 }
             }
-        }
 
-        // If mapped property exists and is not configured => let do the MappedPropertyInputEnricher the enrichment
-        if(mappedProp != null) {
-            if(input.configuration == null || !input.configuration.getProperties().containsKey(mappedProp.getName())) {
-                return enrichMappedProperty(inputColumnProp, mappedProp, input);
+            // If mapped property exists and is not configured => let do the MappedPropertyInputEnricher the enrichment
+            if(mappedProp != null) {
+                if(input.configuration == null || !input.configuration.getProperties().containsKey(mappedProp.getName())) {
+                    return enrichMappedProperty(inputColumnProp, mappedProp, input);
+                }
             }
         }
 
+        if(enrichMapDataObjectsToInputColumnsByName(inputColProperties, input)) {
+            return true;
+        }
+
         enrichWithSingleInputProperty(inputColumnProp, input);
         return true;
     }
 
+    private boolean enrichMapDataObjectsToInputColumnsByName(List<ConfiguredPropertyDescriptor> inputColProperties, ProcessStatelessInput input) {
+        if(inputColProperties.isEmpty()) {
+            return false;
+        }
+        // check if we can convert input "map" to the input columns
+        if(input.data.isArray()) {
+            for(JsonNode item: input.data) {
+                if(!item.isObject()) {
+                    return false;
+                }
+            }
+            // OK, we have array of objects
+        } else if(input.data.isObject()) {
+            // OK, we have one object
+        } else {
+            return false;
+        }
+
+        JsonNodeFactory json = Serializator.getJacksonObjectMapper().getNodeFactory();
+        if(input.data.isArray()) {
+            int i = 0;
+            for(JsonNode item: input.data) {
+                ((ArrayNode)input.data).set(i++, mapDataObjectToInputColumnsByName(item, inputColProperties));
```

Comment:
Just making sure: Is it safe to mutate an ArrayNode you're traversing? I guess so, since you're not adding or deleting elements.

BTW: If you need an indexer, why not use a traditional for loop? This is like the worst of both worlds.